File: Stage21/Java/Problem2565.java
package Stage21.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Problem2565 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        int[][] wire = new int[n][2];
        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            wire[i][0] = Integer.parseInt(st.nextToken());
            wire[i][1] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(wire, (a, b) -> a[0] - b[0]);

        int[] dp = new int[n];
        Arrays.fill(dp, 1);

        int lis = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < i; j++) {
                if (wire[j][1] < wire[i][1]) {
                    dp[i] = Math.max(dp[i], dp[j] + 1);
                }
            }
            lis = Math.max(lis, dp[i]);
        }

        System.out.println(n - lis);
        br.close();
    }
}

File: Stage21/Python/Problem2565.py
def main():
    n = int(input())
    wire = []
    
    for _ in range(n):
        a, b = map(int, input().split())
        wire.append((a, b))
        
    wire.sort()
    dp = [1] * n
    
    for i in range(n):
        for j in range(i):
            if wire[j][1] < wire[i][1]:
                dp[i] = max(dp[i], dp[j] + 1)
    
    print(n - max(dp))
    
if __name__ == "__main__":
    main()